### DISCUSSION TITLE
Training Epochs

### DISCUSSION THREAD

**Discussion by Quiet-Joker** [d8ahazard/sd_dreambooth_extension#323] [Q&A] [answered]
How do Epochs work? i know an Epoch is a full cycle of a dataset. But when i go on the extension and i have 47 images on my dataset and say i want to train 4 Epochs, it always just gives me:
Num examples = 47
  Num batches each epoch = 5
  Num Epochs = 950
  Shuffle After Epoch = False
  Instantaneous batch size per device = 11
  
  a different amount of Epoch number from what i stated. For example, i have 47 and i want to train each images to 101 steps and that would make it 4747 steps in total for 1 Epoch. But how do i go about and do that? There no select Epoch batch size or number of epochs that i can select. Just that `Training Epochs` but no matter what number i put there it always give me another number, I really need a good guide for Dreambooth Training.

**Comment by d8ahazard** [comment, marked as answer]
So, an epoch is just the number of times a given set of instance images will be trained. If you have 100 images and set it to train 1000 steps, then you will wind up with 10 epochs. 

But, now that I'm looking at it, the way it's *supposed* to work is that if you set the number of max steps to None, then it calculates the steps based on the number of epochs provided. 

Of course, there *was* no way to do that in the UI. So, I just pushed an update that will do this if you set the max steps to 0. Set max steps to 0, set epochs to 100 or whatever, and just train. 

What's convenient is, this will work uniformly for all concepts, so the "stop concept after n steps" bit is really not needed...

**Reply by Quiet-Joker** [reply]
Nice, thank you!

**Comment by FugueSegue** [comment]
I'm looking at this right now (c2fc683). I like the idea of calculating the number of steps based on a specified number of epochs. It makes sense.

What is the purpose of the "Maximum Training Steps" text box in the Concepts tabs? I see the default value is -1. There is no hover-text for those text boxes. If the number of epochs dictates the number of training steps for all the concepts, then I agree there is no need for it.
